ml-conversational-analytic-tool/run.py

```diff
@@ -16,20 +15,18 @@ def run(annotated_filename, dataset_filename, outcome, encoding_type, model_type
     data.setupPreProcess(annotated_filename, dataset_filename)
     data.encodeData()
 
-    # Get data for training
-    if encoding_type == 'role':
-        obs, res = data.getRoleMatrix(outcome, padding)
-    elif encoding_type == 'role-agnostic':
-        obs, res = data.getRoleAgnosticMatrix(outcome, padding)
-
     # Create models
-    if model_type == 'CNN':
-        model = BaseCNN()
-    elif model_type == 'LSTM':
+    if model_type == 'LSTM':
```

Comment:
Since i am still trying to understand the code base, Can i know the intention behind this change?

Reply:
When a user passes a model type that is not 'CNN' or 'LSTM', it throws variable reference before assignment error. Therefore, I made CNN the default by removing `elif` to make sure the `model` is always initialized. I also added input validation to make sure that users can pass in only CNN and LSTM.

Reply:
I think that setting a model default value should be in the argument definition:
`parser.add_argument('model', default='CNN', help='Model type to use for training')`
It is more straightforward and the code is clean.

Reply:
Sorry for the confusion, should have not used the word default. I don't want to set CNN to default. The user should always explicitly pass in what algorithm they want to use with the flags that are appropriate for that training. 

With the input validation, else statement will always be CNN.

Reply:
a, ok - didn't get that logic. So, in that case may be it is better to extend the validation - to exit with an error if the model parameter is missing ?

Reply:
I see, so in that case model must be either CNN or LSTM in run(). Thanks!